### PR TITLE
[MMCA-4401]  Accessing bookmarked the page without deliverable email address

### PR DESCRIPTION
**Pull request by wg-hmrc**
[Todo]
- Refactor the `P` component
- Add more unit tests

### CODE REVIEW

File: app/controllers/actions/EmailAction.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                             val messagesApi: MessagesApi)
+  extends ActionFilter[IdentifierRequest] with I18nSupport {
+
+  def filter[A](request: IdentifierRequest[A]): Future[Option[Result]] = {
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
+    dataStoreConnector.getEmail(request.eori).map {
+      case Left(value) => checkEmailResponseAndRedirect(value)
+      case Right(_) => None
+    }.recover { case _ =>
+        None
```
Comment (AdamCoops) [inline app/controllers/actions/EmailAction.scala:43]: Do we need to log an error or anything here?

File: conf/messages.cy [hunk 2]
```diff
@@ -113,6 +113,21 @@ cf.error.not-found.message.go-to-home-page-text1=Neu gallwch
...
 
+# Undeliverable Email Address
+# ----------------------------------------------------------
+cf.undeliverable.email.title=Mae problem gyda’ch cyfeiriad e-bost ar gyfer y Gwasanaeth Datganiadau Tollau
+cf.undeliverable.email.heading=Mae problem gyda’ch cyfeiriad e-bost ar gyfer y Gwasanaeth Datganiadau Tollau
+cf.undeliverable.email.p1=Gwnaethom geisio anfon e-bost atoch ond nid oedd modd ei ddosbarthu.
+cf.undeliverable.email.p2=Eich e-bost cofrestredig yw <strong class="bold">{0}</strong>.
+
+cf.undeliverable.email.verify.heading=Cadarnhau’ch cyfeiriad e-bost
+cf.undeliverable.email.verify.text.p1=Os mae’r cyfeiriad e-bost hwn yn gywir, bydd angen i chi ei gadarnhau er mwyn parhau i ddefnyddio Gwasanaeth Datganiadau Tollau.
```
Comment (AdamCoops) [inline conf/messages.cy:124]: I noticed your double '' in the English version as I assume the single ' doesn't display? Do we need to double them in the welsh as well?

**Comment by AdamCoops** [comment]
It looks like some of the commits haven't been signed as well which will stop the PR from being merged.

File: conf/messages.cy (hunk 2 above)
Comment (harshare) [reply conf/messages.cy:124]: @AdamCoops Single quotes used in English messages file and welsh file are different. Welsh translation is working with a single quote in the messages file.
image

File: app/controllers/actions/EmailAction.scala (hunk 1 above)
Comment (harshare) [reply app/controllers/actions/EmailAction.scala:47]: @AdamCoops Added an error log

**Review by AdamCoops — APPROVED** [review]

**Comment by wg-hmrc** [comment]
> It looks like some of the commits haven't been signed as well which will stop the PR from being merged.

Thanks @AdamCoops regarding  commits haven't been signed, @muhammadibraaheem  just needs to fix

File: app/controllers/actions/EmailAction.scala (hunk 1 above)
Comment (wg-hmrc) [reply app/controllers/actions/EmailAction.scala:47]: this will be done in another ticket

File: conf/messages.cy (hunk 2 above)
Comment (wg-hmrc) [reply conf/messages.cy:124]: double quotes are required in Html classes

**Pull request closed without merging**